packages/indexer-agent/src/query-fees/allocations.ts

```diff
@@ -194,14 +203,104 @@ export class AllocationReceiptCollector implements ReceiptCollector {
   private startVoucherProcessing() {
     timer(30_000).pipe(async () => {
       const pendingVouchers = await this.pendingVouchers()
-      for (const voucher of pendingVouchers) {
-        await this.submitVoucher(voucher)
+
+      const logger = this.logger.child({})
+
+      const vouchers = await pReduce(
+        pendingVouchers,
+        async (results, voucher) => {
+          if (
+            await this.allocationExchange.allocationsRedeemed(
+              voucher.allocation,
+            )
+          ) {
+            logger.warn(
+              `Query fee voucher for allocation already redeemed, deleted local voucher copy`,
+              { allocation: voucher.allocation },
+            )
+            try {
+              await this.models.vouchers.destroy({
+                where: { allocation: voucher.allocation },
+              })
+            } catch (err) {
+              logger.warn(
+                `Failed to delete local vouchers copy, will try again later`,
+                { err, allocation: voucher.allocation },
+              )
+            }
+            return results
+          }
+          if (
+            BigNumber.from(voucher.amount).lt(this.voucherRedemptionThreshold)
+          ) {
+            results.belowThreshold.push(voucher)
+          } else {
+            results.eligible.push(voucher)
+          }
+          return results
+        },
+        { belowThreshold: <Voucher[]>[], eligible: <Voucher[]>[] },
```

Comment:
Why are these types `<Voucher[]>[]` rather than `Voucher[]`?

Reply:
It's a type-annotated assignment (i.e. it sets the value to `[]`). Replacing it with `Voucher[]` yields `Element implicitly has an 'any' type because expression of type 'any' can't be used to index type 'typeof Voucher'.ts(7053)`, because it's expecting a value not a type.

Reply:
👍